[2025-06-27T05:05:42.568+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-06-27T05:05:42.619+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_processing_pipeline.Structure_Data_With_Spark manual__2025-06-27T05:05:35+00:00 [queued]>
[2025-06-27T05:05:42.627+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_processing_pipeline.Structure_Data_With_Spark manual__2025-06-27T05:05:35+00:00 [queued]>
[2025-06-27T05:05:42.628+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-06-27T05:05:42.667+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): Structure_Data_With_Spark> on 2025-06-27 05:05:35+00:00
[2025-06-27T05:05:42.673+0000] {standard_task_runner.py:64} INFO - Started process 204 to run task
[2025-06-27T05:05:42.676+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spark_processing_pipeline', 'Structure_Data_With_Spark', 'manual__2025-06-27T05:05:35+00:00', '--job-id', '23', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmp48xddlnj']
[2025-06-27T05:05:42.679+0000] {standard_task_runner.py:91} INFO - Job 23: Subtask Structure_Data_With_Spark
[2025-06-27T05:05:42.810+0000] {task_command.py:426} INFO - Running <TaskInstance: spark_processing_pipeline.Structure_Data_With_Spark manual__2025-06-27T05:05:35+00:00 [running]> on host f7d2d893e18e
[2025-06-27T05:05:43.208+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_processing_pipeline' AIRFLOW_CTX_TASK_ID='Structure_Data_With_Spark' AIRFLOW_CTX_EXECUTION_DATE='2025-06-27T05:05:35+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-27T05:05:35+00:00'
[2025-06-27T05:05:43.209+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-06-27T05:05:43.248+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-06-27T05:05:43.249+0000] {spark_submit.py:401} INFO - Spark-Submit cmd: /home/***/.local/lib/python3.10/site-packages/pyspark/bin/spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --queue root.default /opt/***/include/scripts/structured_data.py /opt/***/data/raw/raw.json /opt/***/data/foundation
[2025-06-27T05:05:43.267+0000] {spark_submit.py:571} INFO - /home/***/.local/lib/python3.10/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-06-27T05:05:43.268+0000] {spark_submit.py:571} INFO - JAVA_HOME is not set
[2025-06-27T05:05:43.269+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-06-27T05:05:43.301+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 502, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: /home/***/.local/lib/python3.10/site-packages/pyspark/bin/spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --queue root.default /opt/***/include/scripts/structured_data.py /opt/***/data/raw/raw.json /opt/***/data/foundation. Error code is: 1.
[2025-06-27T05:05:43.304+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_processing_pipeline, task_id=Structure_Data_With_Spark, run_id=manual__2025-06-27T05:05:35+00:00, execution_date=20250627T050535, start_date=20250627T050542, end_date=20250627T050543
[2025-06-27T05:05:43.383+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 23 for task Structure_Data_With_Spark (Cannot execute: /home/***/.local/lib/python3.10/site-packages/pyspark/bin/spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --queue root.default /opt/***/include/scripts/structured_data.py /opt/***/data/raw/raw.json /opt/***/data/foundation. Error code is: 1.; 204)
[2025-06-27T05:05:43.410+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-06-27T05:05:43.451+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-27T05:05:43.452+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
